File: app/src/main/java/com/a65apps/yuhnin/lesson1/ui/adapters/ContactListAdapter.java
package com.a65apps.yuhnin.lesson1.ui.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;

import com.a65apps.yuhnin.lesson1.R;
import com.a65apps.yuhnin.lesson1.pojo.ContactInfoModel;
import java.util.List;

public class ContactListAdapter  extends BaseAdapter {
    List<ContactInfoModel> contactList;
    Context context;

    public ContactListAdapter(Context context, List<ContactInfoModel> contactList){
        this.context = context;
        this.contactList = contactList;
    }

    @Override
    public int getCount() {
        return contactList.size();
    }

    @Override
    public Object getItem(int position) {
        return contactList.get(position);
    }

    @Override
    public long getItemId(int position) {
        return contactList.get(position).getId();
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            LayoutInflater inflater = LayoutInflater.from(context);
            convertView = inflater.inflate(R.layout.layout_contact_list_item, parent, false);
            ImageView ivContactType = convertView.findViewById(R.id.iv_contacts_type);
            TextView tvContactType = convertView.findViewById(R.id.tv_contacts_type);
            TextView tvValue = convertView.findViewById(R.id.tv_contacts_value);
            switch (contactList.get(position).getContactType()) {
                case EMAIL:
                    ivContactType.setImageResource(R.drawable.typeemail);
                    tvContactType.setText("Email:");
                    break;
                case PHONE_NUMBER:
                    ivContactType.setImageResource(R.drawable.typephone);
                    tvContactType.setText("Телефон:");
                    break;
            }
            tvValue.setText(contactList.get(position).getValue());
        }
        return convertView;
    }
}

File: library/src/main/java/com/a65apps/library/ui/adapters/ContactListAdapter.kt
package com.a65apps.library.ui.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.a65apps.core.entities.ContactType
import com.a65apps.library.R
import com.a65apps.library.models.ContactModel
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.layout_contact_list_item.*


class ContactListAdapter(private val contactList: List<ContactModel>) :
        RecyclerView.Adapter<ContactListAdapter.ContactViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ContactViewHolder {
        val view: View = LayoutInflater.from(parent.context)
                .inflate(R.layout.layout_contact_list_item, parent, false)
        return ContactViewHolder(view)
    }

    override fun onBindViewHolder(holder: ContactViewHolder, position: Int) {
        holder.bind(contactList[position])
    }

    override fun getItemCount() = contactList.size

    class ContactViewHolder(override val containerView: View)
        : RecyclerView.ViewHolder(containerView), LayoutContainer {
        fun bind(contact: ContactModel) {
            when (contact.contactType) {
                ContactType.EMAIL -> {
                    imageviewContactsType.setImageResource(R.drawable.typeemail)
                    textviewContactsType.text = containerView.context.getText(R.string.text_contact_type_email)
                }
                ContactType.PHONE_NUMBER -> {
                    imageviewContactsType.setImageResource(R.drawable.typephone)
                    textviewContactsType.text = containerView.context.getText(R.string.text_contact_type_phone)
                }
                ContactType.SKYPE -> {
                    imageviewContactsType.setImageResource(R.drawable.typeskype)
                    textviewContactsType.text = containerView.context.getText(R.string.text_contact_type_skype)
                }
                ContactType.WEBSITE -> {
                    imageviewContactsType.setImageResource(R.drawable.typesite)
                    textviewContactsType.text = containerView.context.getText(R.string.text_contact_type_site)
                }
                else -> {
                    imageviewContactsType.setImageResource(R.drawable.typesite)
                    textviewContactsType.text = containerView.context.getText(R.string.text_notset)
                }
            }
            textviewContactsValue.text = contact.value
        }
    }
}